tests/php/Security/SecurityTest.php

```diff
@@ -548,6 +551,80 @@ public function testChangePasswordFromLostPassword()
+            // anonymous class without instantiating it unfortunately.
+            new class('', new MemberAuthenticator()) extends ChangePasswordHandler {
+                private static $allowed_actions = [
+                    'changepassword',
+                ];
+                protected function createChangePasswordResponse(): array|HTTPResponse
+                {
+                    // This is an invalid response, and is included to validate that the valid handler will be used
+                    // even if there's an invalid one in the group
+                    return [];
+                }
+            }
+        );
+        $secondaryAuthenticator = new class($secondaryPasswordHandlerClass) extends MemberAuthenticator {
+            public function __construct(private string $handlerClass)
+            {
+            }
+            public function getChangePasswordHandler($link)
+            {
+                return $this->handlerClass::create($link, $this);
+            }
+        };
+        $security = Security::singleton();
+        try {
+            // Add the dummy authenticator and prep request
+            $authenticators = $security->getAuthenticators();
+            $authenticators['test-auth'] = $secondaryAuthenticator;
+            $security->setAuthenticators($authenticators);
+            $params = [
+                't' => 'some-token',
+                'm' => $member->ID,
+            ];
+            // Don't include "/Security" because that would have been routed out by Security admin by now
+            $request = new HTTPRequest('GET', '/changepassword', $params);
+            $request->setSession(new Session([]));
+            $security->setRequest($request);
+
+            // Initial request should redirect to the form with a new temp hash
+            $response = $security->changepassword();
+            $this->assertInstanceOf(HTTPResponse::class, $response);
+            $this->assertSame(302, $response->getStatusCode());
+            $locationPrefix = Director::absoluteURL('/Security/changepassword?th=');
+            $actualLocation = $response->getHeader('Location');
+            $this->assertStringStartsWith($locationPrefix, $actualLocation);
+
+            // Second request should return the form
+            $params = [
+                'th' => str_replace($locationPrefix, '', $actualLocation),
+            ];
+            $request = new HTTPRequest('GET', '/changepassword', $params);
+            $request->setSession(new Session([]));
+            $security->setRequest($request);
+            $response = $security->changepassword();
+            $this->assertInstanceOf(DBHTMLText::class, $response);
+            $this->assertStringNotContainsString('The password reset link is invalid or expired', $response->getValue());
+            $this->assertStringContainsString('Please enter a new password', $response->getValue());
```

Comment:
I would have checked for the presence of the form, except there isn't one due to https://github.com/silverstripe/startup-theme/issues/17

Not sure how other tests in here that rely on the form are passing :sweat_smile: